### ISSUE TITLE
Unsupported operation: Infinity or NaN toInt

### ISSUE THREAD

**Issue by MichaelTamm** [flutter/flutter#77908]
The following error was reported by an Android app (currently in internal testing) using Flutter 2.0.1 and Dart 2.0.12, release build with null-safety:

```
UnsupportedError: Unsupported operation: Infinity or NaN toInt
  File "double.dart", line 219, in _Double.toInt
  File "semantics.dart", line 2328, in _BoxEdge.compareTo
  File "comparable.dart", line 88, in Comparable.compare
  File "list.dart", line 359, in ListMixin._compareAny
  File "sort.dart", line 69, in Sort._insertionSort
  File "sort.dart", line 58, in Sort._doSort
  File "sort.dart", line 33, in Sort.sort
  File "list.dart", line 355, in ListMixin.sort
  File "semantics.dart", line 2523, in _childrenInDefaultOrder
  File "semantics.dart", line 2117, in SemanticsNode._childrenInTraversalOrder
  File "semantics.dart", line 2055, in SemanticsNode._addToUpdate
  File "semantics.dart", line 2653, in SemanticsOwner.sendSemanticsUpdate
  File "object.dart", line 1083, in PipelineOwner.flushSemantics
  File "binding.dart", line 458, in RendererBinding.drawFrame
  File "binding.dart", line 874, in WidgetsBinding.drawFrame
  File "binding.dart", line 319, in RendererBinding._handlePersistentFrameCallback
  File "binding.dart", line 1144, in SchedulerBinding._invokeFrameCallback
  File "binding.dart", line 1082, in SchedulerBinding.handleDrawFrame
  File "binding.dart", line 998, in SchedulerBinding._handleDrawFrame
  File "zone.dart", line 1354, in _rootRun
  File "zone.dart", line 1258, in _CustomZone.run
  File "zone.dart", line 1162, in _CustomZone.runGuarded
  File "hooks.dart", line 161, in _invoke
  File "platform_dispatcher.dart", line 253, in PlatformDispatcher._drawFrame
  File "hooks.dart", line 120, in _drawFrame
```

**Comment by MichaelTamm** [comment]
After a short look into the Flutter source code it looks like a `_BoxEdge` was constructed with an invalid `offset` -- Unfortunately the stack trace does not tell us, where the invalid `_BoxEdge` instance has been created.

**Comment by TahaTesser** [comment]
Hi @MichaelTamm 
is there a way to reproduce this locally?  
Can you please provide your `flutter doctor -v` and `flutter run -release --verbose` logs, and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)?
Thank you

**Comment by MichaelTamm** [comment]
Output of `flutter doctor -v`:
```
[✓] Flutter (Channel stable, 2.0.1, on Linux, locale de_DE.UTF-8)
    • Flutter version 2.0.1 at /home/Michael/flutter.stable
    • Framework revision c5a4b4029c (vor 7 Tagen), 2021-03-04 09:47:48 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/Michael/Android/Sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /usr/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_282-b08)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] VS Code (version 1.54.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.20.0

[✓] Connected device (2 available)
    • SM G930F (mobile) • ce041824782daa3c02 • android-arm64  • Android 8.0.0 (API 26)
    • Chrome (web)      • chrome             • web-javascript • Google Chrome 89.0.4389.82
```
Note: I have Android Studio installed via Jetbrains Toolbox, but that's another issue (https://github.com/flutter/flutter/issues/75590)

**Comment by MichaelTamm** [comment]
I would really like to provide more help for the triage. Right now I cannot run the Flutter App in release mode:
```
$ flutter run --release --flavor local
Launching lib/main.dart on SM G930F in release mode...
                                                                        
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Execution failed for task ':app:lintVitalLocalRelease'.                 
> Could not resolve all artifacts for configuration ':app:prodReleaseRuntimeClasspath'.
   > Failed to transform libs.jar to match attributes {artifactType=processed-jar, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for JetifyTransform: /home/Michael/src/autovio-dev/build/app/intermediates/flutter/prodRelease/libs.jar.
         > Transform's input file does not exist: /home/Michael/src/autovio-dev/build/app/intermediates/flutter/prodRelease/libs.jar. (See https://issuetracker.google.com/issues/158753935)
                                                                        
* Try:                                                                  
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
                                                                        
* Get more help at https://help.gradle.org                              
                                                                        
BUILD FAILED in 55s                                                     
Running Gradle task 'assembleLocalRelease'...                           
Running Gradle task 'assembleLocalRelease'... Done                 56,4s
Exception: Gradle task assembleLocalRelease failed with exit code 1
```
In debug mode everything works fine and there aren't any assertion errors.

**Comment by MichaelTamm** [comment]
The device, from which the Exception orginated was a Nexus 5X (Model: MTC20F) running Android 6.0.1

**Comment by andyduke** [comment]
I faced this problem too. In the Crashlytics log, I see such a trace, but it is not clear from it where the problem begins:
```
Non-fatal Exception: io.flutter.plugins.firebase.crashlytics.FlutterError: Unsupported operation: Infinity or NaN toInt. Error thrown Instance of 'ErrorDescription'.
       at _Double.round(_Double.java)
       at RenderSliverFixedExtentBoxAdaptor.getMinChildIndexForScrollOffset(RenderSliverFixedExtentBoxAdaptor.java:70)
       at RenderSliverFixedExtentBoxAdaptor.performLayout(RenderSliverFixedExtentBoxAdaptor.java:192)
       at RenderObject.layout(RenderObject.java:1858)
       at RenderSliverEdgeInsetsPadding.performLayout(RenderSliverEdgeInsetsPadding.java:137)
       at _RenderSliverFractionalPadding.performLayout(_RenderSliverFractionalPadding.java:167)
       at RenderObject.layout(RenderObject.java:1858)
       at RenderViewportBase.layoutChildSequence(RenderViewportBase.java:510)
       at RenderViewport._attemptLayout(RenderViewport.java:1580)
       at RenderViewport.performLayout(RenderViewport.java:1489)
       at RenderObject._layoutWithoutResize(RenderObject.java:1713)
       at PipelineOwner.flushLayout(PipelineOwner.java:885)
       at RendererBinding.drawFrame(RendererBinding.java:453)
       at WidgetsBinding.drawFrame(WidgetsBinding.java:883)
       at RendererBinding._handlePersistentFrameCallback(RendererBinding.java:319)
       at SchedulerBinding._invokeFrameCallback(SchedulerBinding.java:1143)
       at SchedulerBinding.handleDrawFrame(SchedulerBinding.java:1080)
       at SchedulerBinding._handleDrawFrame(SchedulerBinding.java:996)
```

**Comment by goderbauer** [comment]
Without information of how to reproduce this crash we can't really debug this.
